### PR TITLE
- Created CarDao, CarService, DriverDao, DRiverService

### PR DESCRIPTION
**Pull request by NikolayFatkullin**

### CODE REVIEW

File: src/main/java/com/shop/db/Storage.java [hunk 1]
```diff
@@ -1,16 +1,34 @@
...
 import java.util.ArrayList;
 import java.util.List;
 
 public class Storage {
     public static final List<Manufacturer> manufacturers = new ArrayList<>();
+    public static final List<Car> cars = new ArrayList<>();
+    public static final List<Driver> drivers = new ArrayList<>();
     public static long manufacturerId = 0;
+    public static long carId = 0;
+    public static long driverId = 0;
```
Comment (semyonich) [inline src/main/java/com/shop/db/Storage.java:16]: id field type should be Long, not long

**Review by semyonich — APPROVED** [review]

File: src/main/java/com/shop/db/Storage.java [hunk 2]
```diff
@@ -1,16 +1,34 @@
...
```
Comment (RomanDubovskyi) [inline src/main/java/com/shop/db/Storage.java:15]:
```suggestion
    public static Long carId = 0;
    public static Long driverId = 0;
```

File: src/main/java/com/shop/service/impl/CarServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+    @Override
+    public void addDriverToCar(Driver driver, Car car) {
+        if (driverDao.get(driver.getId()).isEmpty()) {
+            driverDao.create(driver);
+        }
+        car.getDrivers().add(driver);
+        if (carDao.get(car.getId()).isEmpty()) {
+            carDao.create(car);
+        } else {
```
Comment (RomanDubovskyi) [inline src/main/java/com/shop/service/impl/CarServiceImpl.java:53]: we don't need these check in service, db will handle this case and throw exception if there is no driver in the db

File: src/main/java/com/shop/service/impl/CarServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+            carDao.update(car);
+        }
+    }
+
+    @Override
+    public void removeDriverFromCar(Driver driver, Car car) {
+        carDao.get(car.getId()).get().getDrivers().removeIf(d -> d.equals(driver));
```
Comment (RomanDubovskyi) [inline src/main/java/com/shop/service/impl/CarServiceImpl.java:60]: samec here
and don't forget to call update after

File: src/main/java/com/shop/service/impl/CarServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        return carDao.getAll().stream()
+                .filter(c -> c.getDrivers()
+                        .stream().anyMatch(d -> d.getId().equals(driverId)))
+                .collect(Collectors.toList());
```
Comment (RomanDubovskyi) [inline src/main/java/com/shop/service/impl/CarServiceImpl.java:68]: let's move this logic to the dao layer, we should received already filtered data from db

**Review by RomanDubovskyi — CHANGES_REQUESTED** [review]

File: src/main/java/com/shop/service/impl/CarServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+    }
+
+    @Override
+    public void addDriverToCar(Driver driver, Car car) {
+        car.getDrivers().add(driver);
+        if (carDao.get(car.getId()).isEmpty()) {
+            carDao.create(car);
+        } else {
+            carDao.update(car);
+        }
```
Comment (RomanDubovskyi) [inline src/main/java/com/shop/service/impl/CarServiceImpl.java:51]:
```suggestion
        car.getDrivers().add(driver);
        carDao.update(car);
```

File: src/main/java/com/shop/service/impl/CarServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+    }
+
+    @Override
+    public void removeDriverFromCar(Driver driver, Car car) {
+        carDao.get(car.getId()).get().getDrivers().removeIf(d -> d.equals(driver));
```
Comment (RomanDubovskyi) [inline src/main/java/com/shop/service/impl/CarServiceImpl.java:56]: same here

**Review by RomanDubovskyi — CHANGES_REQUESTED** [review]
great! 
almost there

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request merged**